src/codegen/llvm.zig

```diff
@@ -4320,79 +4320,40 @@ pub const FuncGen = struct {
         return self.builder.buildCall(fn_val, &params, params.len, .C, .Auto, "");
     }
 
-    fn airClzCtz(self: *FuncGen, inst: Air.Inst.Index, prefix: [*:0]const u8) !?*const llvm.Value {
+    fn airBitCount(
+        self: *FuncGen,
+        inst: Air.Inst.Index,
+        llvm_fn_name: []const u8,
+        zero_poison_flag: bool,
+    ) !?*const llvm.Value {
         if (self.liveness.isUnused(inst)) return null;
 
         const ty_op = self.air.instructions.items(.data)[inst].ty_op;
         const operand_ty = self.air.typeOf(ty_op.operand);
         const operand = try self.resolveInst(ty_op.operand);
         const target = self.dg.module.getTarget();
-        const bits = operand_ty.intInfo(target).bits;
-        const vec_len: ?u32 = switch (operand_ty.zigTypeTag()) {
-            .Vector => operand_ty.vectorLen(),
-            else => null,
-        };
 
-        var fn_name_buf: [100]u8 = undefined;
-        const llvm_fn_name = if (vec_len) |len|
-            std.fmt.bufPrintZ(&fn_name_buf, "llvm.{s}.v{d}i{d}", .{
-                prefix, len, bits,
-            }) catch unreachable
-        else
-            std.fmt.bufPrintZ(&fn_name_buf, "llvm.{s}.i{d}", .{
-                prefix, bits,
-            }) catch unreachable;
-        const llvm_i1 = self.context.intType(1);
-        const fn_val = self.dg.object.llvm_module.getNamedFunction(llvm_fn_name) orelse blk: {
-            const operand_llvm_ty = try self.dg.llvmType(operand_ty);
-            const param_types = [_]*const llvm.Type{ operand_llvm_ty, llvm_i1 };
-            const fn_type = llvm.functionType(operand_llvm_ty, &param_types, param_types.len, .False);
-            break :blk self.dg.object.llvm_module.addFunction(llvm_fn_name, fn_type);
-        };
+        var params = std.BoundedArray(*const llvm.Value, 2).init(0) catch unreachable;
+        params.append(operand) catch unreachable;
+        if (zero_poison_flag) {
```

Comment:
could be nice to explain briefly the zero_poison_flag, which applies on some targets for `input=0` for `clz` and `ctz` due to how the hardware works ie `//assert: input != 0 for some targets`.
Or do you think that this is self-evident from reading `compiler_rt`?

Reply:
Documentation for this is found in the doc comments for the AIR instruction of ctz and clz, as well as the LLVM language reference.